cli/azd/pkg/project/service.go

```diff
@@ -76,6 +78,12 @@ func (svc *Service) Deploy(
 			return
 		}
 
+		// Allow users to specify their own endpoints, in cases where they've configured their own front-end load balancers,
+		// reverse proxies or DNS host names outside of the service target (and prefer that to be used instead).
+		if endpointsOverride := env.GetServiceProperty(svc.Config.Name, "ENDPOINTS"); endpointsOverride != "" {
```

Comment:
What do we think about this solution overall? Ultimately, we want the user to be able to easily query the infrastructure setup and provide the result. `bicep` provides the user with ability to do just that, and the passthrough of result happens via the environment file currently.

Also note that this override is provided not at the presentation layer, but at the business logic layer. `azd show` will use the overridden endpoints if available.

Reply:
This feels like the right direction to me.  I think it aligns well with the existing model.

I am wondering if it makes sense for us to say that the output type should be an array of strings instead of just a string that we break apart.  This feels like it's slightly more idiomatic in bicep land, and it would be more pleasant to consume that output in a bicep context.

I think the major problem with this approach is that we don't have a great way of storing non string values in the environment today since there's a tight correspondence between them and environment variables.  On some level it feels "wrong" that these properties (both the endpoints, and the image name, for containers) leak into the environment variables, but this is the best technology we have today.

Maybe sticking with string for now is the right idea, if we want to expand the support later to also handle array properties, we should be able to do that in a compatible way (and perhaps deprecate the old way at that time).

Mostly rambling here - but wanted to give you an idea of where my head is at.  To be clear, I think this is a great step forward and we should continue with it.

Reply:
nit.
I would rename `endpointsOverride` to `overriddenEndpoints`.  The former reads as a Boolean, but it is a list